prov/rxm/src/rxm_cq.c

```diff
@@ -179,21 +177,14 @@ static int rxm_finish_recv(struct rxm_rx_buf *rx_buf, size_t done_len)
 			rxm_cntr_incerr(rx_buf->ep->util_ep.rx_cntr);
 	} else {
 		if (rx_buf->recv_entry->flags & FI_COMPLETION) {
-			FI_DBG(&rxm_prov, FI_LOG_CQ, "writing recv completion: "
-			       "length: %" PRIu64 ", tag: 0x%" PRIx64 "\n",
-			       rx_buf->pkt.hdr.size, rx_buf->pkt.hdr.tag);
-			ret = ofi_cq_write(rx_buf->ep->util_ep.rx_cq,
-					   rx_buf->recv_entry->context,
-					   rx_buf->recv_entry->comp_flags |
-					   rx_buf->pkt.hdr.flags,
-					   rx_buf->pkt.hdr.size,
-					   rx_buf->recv_entry->rxm_iov.iov[0].iov_base,
-					   rx_buf->pkt.hdr.data, rx_buf->pkt.hdr.tag);
-			if (OFI_UNLIKELY(ret)) {
-				FI_WARN(&rxm_prov, FI_LOG_CQ,
-					"Unable to write recv completion\n");
+			ret = rxm_cq_write_recv_comp(rx_buf,
+						     rx_buf->recv_entry->context,
+						     (rx_buf->recv_entry->comp_flags |
+						      rx_buf->pkt.hdr.flags),
+						     rx_buf->pkt.hdr.size,
+						     NULL);
```

Comment:
It has to be `rx_buf->recv_entry->rxm_iov.iov[0].iov_base` instead of `NULL`.

For example, MPICH/CH4 uses `cq_entry->buf` in Active Messages implementation.

Reply:
Active messages use `FI_MULTI_RECV`, right? I think I can check for the flag to selectively fill this field, but I can skip it to avoid the branch. The man pages don't forbid reporting the buffer for regular receives.

Reply:
Right, AM use `FI_MULTI_RECV`. It would be great to avoid the branch.
@shefty Could you comment, please? Is it OK to fill this field always?

Reply:
@a-ilango Thanks!

Reply:
It is harmless to fill out this field, but the app should not expect it to point to anything meaningful outside of FI_MULTI_RECV